proxy/context.go

```diff
@@ -287,17 +289,19 @@ func (c *context) Split() (filters.FilterContext, error) {
 		c.Logger().Errorf("context: failed to clone request: %v", err)
 		return nil, err
 	}
-	serverSpan := opentracing.SpanFromContext(originalRequest.Context())
-	cr = cr.WithContext(opentracing.ContextWithSpan(cr.Context(), serverSpan))
+	serverSpan := tracing.SpanFromContext(originalRequest.Context(), c.tracer)
+	cr = cr.WithContext(tracing.ContextWithSpan(cr.Context(), serverSpan))
 	cr = cr.WithContext(routing.NewContext(cr.Context()))
+
 	originalRequest.Body = body
 	cc.request = cr
 	return cc, nil
 }
 
 func (c *context) Loopback() {
-	loopSpan := c.Tracer().StartSpan(c.proxy.tracing.initialOperationName, opentracing.ChildOf(c.ParentSpan().Context()))
-	defer loopSpan.Finish()
+	_, loopSpan := c.Tracer().Start(c.request.Context(), c.proxy.tracing.initialOperationName)
```

Comment:
What return value is ignored here and why?

Reply:
This is an stdlib context. This context doesn't seem to be used in the function.

One thing that could be done is update the request context so it carries the span information forward if any other function or package wants to use it. 

But in theory this was not done in the past either, `Ot.ContextWithSpan(...)` was never called here so the context was never updated, the only difference is that `Otel.Tracer.Start(...)` always returns a context, which was not true for OT implementation.